fhir-model/src/main/java/com/ibm/fhir/model/type/Canonical.java

```diff
@@ -33,6 +33,25 @@ public static Canonical of(java.lang.String value) {
         return Canonical.builder().value(value).build();
     }
 
+    public static Canonical of(java.lang.String uri, java.lang.String version) {
+        StringBuilder value = new StringBuilder(uri);
+        if (version != null && !version.isEmpty()) {
+            value.append("|" + version);
+        }
+        return Canonical.builder().value(value.toString()).build();
+    }
+
+    public static Canonical of(java.lang.String uri, java.lang.String version, java.lang.String fragment) {
```

Comment:
If you are going to use StringBuilder then use it for everything:

value.append("|").append(version);

don't mix and match string builder and the concatenation operator. this will give you the most benefit of using something like string builder

Reply:
done